recipes/apr/all/conanfile.py

```diff
@@ -83,6 +89,56 @@ def build_requirements(self):
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
+    def _get_cross_building_configure_args(self):
+        """
+        The APR configure.in script makes extensive (30 instances) use
+        Check warning on line 73 in recipes/apr/all/conanfile.py
+
+        String statement has no effect of AC_TRY_RUN (to determine system capabilities) and
+        checks for /dev/zero. These runtime checks only work when run on the host so
+        the configure script will fail when cross compiling unless the relevant
+        configuration variables are provided in a cache file or on the command line or in an
+        environment variable. The configuration variable values are most easily determined by
+        running the configure script on a host system using:
```

Comment:
I think the issue is that apr build scripts perform some checks that don't have a fallback when cross-building, and the values need to provided at ./configure time with knowledge of the target system.  I'd try to make this a bit more "compact" of an explanation, but the functionality is good!